src/client/driver/command-executors/execute-action-command.js

```diff
@@ -56,20 +130,47 @@ function ensureElement (selector, timeout, NotFoundErrorCtor, IsInvisibleErrorCt
         });
 }
 
+function getEnsureElementPromise (selector, timeout) {
+    return ensureElement(selector, timeout, ActionElementNotFoundError, ActionElementIsInvisibleError);
+}
+
 function ensureCommandElements (command, timeout) {
     var progressPanel = new ProgressPanel();
 
     progressPanel.show(PROGRESS_PANEL_TEXT, timeout);
 
     var ensureElementPromises = [];
 
-    if (command.selector)
-        ensureElementPromises.push(ensureElement(command.selector, timeout, ActionElementNotFoundError, ActionElementIsInvisibleError));
+    if (command.selector) {
+        var ensureElementPromise = getEnsureElementPromise(command.selector, timeout);
+
+        if (command.type === COMMAND_TYPE.selectText)
+            ensureElementPromises.push(ensureElementPromise.then(el => ensureElementEditable(el)));
+        else if (command.type === COMMAND_TYPE.selectTextAreaContent)
+            ensureElementPromises.push(ensureElementPromise.then(el => ensureTextAreaElement(el)));
+        else
+            ensureElementPromises.push(ensureElementPromise);
+    }
+
+    if (command.type === COMMAND_TYPE.dragToElement) {
+        ensureElementPromises.push(ensureElement(command.destinationSelector, timeout,
+            DragDestinationNotFoundError, DragDestinationIsInvisibleError));
+    }
 
-    if (command.type === COMMAND_TYPE.dragToElement)
-        ensureElementPromises.push(ensureElement(command.destinationSelector, timeout, DragDestinationNotFoundError, DragDestinationIsInvisibleError));
+    if (command.type === COMMAND_TYPE.selectEditableContent) {
+        var endNodeSelector = command.endNodeSelector || command.startNodeSelector;
+
+        ensureElementPromises.push(getEnsureElementPromise(command.startNodeSelector, timeout).then(el => ensureContentEditableElement(el)));
```

Comment:
I suppose it should be different errors for `startNodeSelector` and `andNodeSelector`. Otherwise how do you determine in the error what the selector wrong?